src/python/lib/ensembl/compara/utils/cigar.py

```diff
@@ -0,0 +1,160 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.ls -l
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Collection of utils methods targeted to cigar line manipulation.
+
+Typical usage examples::
+
+
+"""
+
+__all__ = ['aligned_seq_to_cigar', 'alignment_to_seq_coordinate', 'get_cigar_array']
+
+from typing import List, Tuple
+
+def aligned_seq_to_cigar(aligned_seq:str) -> str:
+    """ Convert an aligned_seq to its cigar line
+
+    The cigar line use [legth][operation] order and if the operation length is 1 it give the operation
+    A-TGC---CC --> MD3M3D2M
+
+    Params:
+        aligned_seq: the aligned sequence (str)
+
+    Returns:
+         cigar line (str)
+    """
+    cigar = ""
```

Comment:
As a first suggestion, I'd recommend to build up the CIGAR as a list of strings, as described in the [Python programming FAQ entry on concatenating many strings](https://docs.python.org/3.7/faq/programming.html#what-is-the-most-efficient-way-to-concatenate-many-strings-together), and touched on in the [first point of the PEP 8 programming recommendations](https://peps.python.org/pep-0008/#programming-recommendations). This is a good practice in general, and in the case of a CIGAR representing a genomic alignment, it is likely to yield a significant performance improvement.